ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java

```diff
@@ -183,6 +192,188 @@ static void validateDefaultTopicFormats(final KsqlConfig config) {
     validateTopicFormat(config, KsqlConfig.KSQL_DEFAULT_VALUE_FORMAT_CONFIG, "value");
   }
 
+  @VisibleForTesting
+  static void validateFips(final KsqlConfig config, final KsqlRestConfig restConfig) {
+    if (config.getBoolean(ConfluentConfigs.ENABLE_FIPS_CONFIG)) {
+      final FipsValidator fipsValidator = ConfluentConfigs.buildFipsValidator();
+
+      // validate state store password encryption type
+      validateStateStoreType(restConfig);
+
+      // validate cipher suites and TLS version
+      validateCipherSuites(fipsValidator, restConfig);
+
+      // validate broker
+      validateBroker(fipsValidator, config);
+
+      // validate ssl endpoint algorithm
+      validateSslEndpointAlgo(fipsValidator, restConfig);
+
+      // validate schema registry url
+      validateSrUrl(fipsValidator, restConfig);
+
+      // validate all listeners
+      validateListeners(fipsValidator, restConfig);
+
+      log.info("FIPS mode enabled for ksqlDB!");
+    }
+  }
+
+  private static void validateStateStoreType(final KsqlRestConfig restConfig) {
```

Comment:
I guess this has nothing to do with the state stores used in ksql/Kafka Streams, doesn't it? I would rename this to `validateTrustStoreType()`.
Do you also need to validate key store types?